meilisearch-http/tests/index/create_index.rs

```diff
@@ -18,6 +19,57 @@ async fn create_index_no_primary_key() {
     assert_eq!(response["details"]["primaryKey"], Value::Null);
 }
 
+#[actix_rt::test]
+async fn create_index_with_gzip_encoded_request() {
```

Comment:
Does it already support sending encoded responses when the accept-encoding header is set?
If not, could you please add one or two tests, please?

Reply:
I added two tests for response encoding. Had difficulties with `decode` function. I believe my implementation is not idiomatic (too much copying), so I will really appreciate your suggestions.